esiclient/v1/node_network.py

```diff
@@ -0,0 +1,203 @@
+                                 network.name, fixed_ip])
+            elif not parsed_args.network:
+                data.append([node.name, port.address, None, None])
+
+        return ["Node", "MAC Address", "Network", "Fixed IP"], data
+
+
+class Attach(command.ShowOne):
+    """Attach network to node"""
+
+    log = logging.getLogger(__name__ + ".Attach")
+
+    def get_parser(self, prog_name):
+        parser = super(Attach, self).get_parser(prog_name)
+        parser.add_argument(
+            "node",
+            metavar="<node>",
+            help=_("Name or UUID of the node"))
+        parser.add_argument(
+            "network",
+            metavar="<network>",
+            help=_("Name or UUID of the network"))
+
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)" % parsed_args)
+
+        node_uuid = parsed_args.node
+        network_uuid = parsed_args.network
+
+        ironic_client = self.app.client_manager.baremetal
+        neutron_client = self.app.client_manager.network
+
+        node = ironic_client.node.get(node_uuid)
+        network = neutron_client.find_network(network_uuid)
+
+        if node.provision_state == MANAGEABLE:
+            # adopt the node
+            node_update = []
+            node_revert = []
+            if 'image_source' not in node.instance_info:
+                temp_image = node.driver_info['deploy_ramdisk']
+                node_update.append({'path': '/instance_info/image_source',
+                                    'value': temp_image,
+                                    'op': 'add'})
+                node_revert.append({'path': '/instance_info/image_source',
+                                    'op': 'remove'})
+            if 'capabilities' not in node.instance_info:
+                node_update.append({'path': '/instance_info/capabilities',
+                                    'value': "{\"boot_option\": \"local\"}",
+                                    'op': 'add'})
+                node_revert.append({'path': '/instance_info/capabilities',
+                                    'op': 'remove'})
+
+            if len(node_update) > 0:
+                ironic_client.node.update(node_uuid, node_update)
+            ironic_client.node.set_provision_state(node_uuid, ADOPT)
+            if len(node_revert) > 0:
+                ironic_client.node.update(node_uuid, node_revert)
+            # reload node information
+            node = ironic_client.node.get(node_uuid)
+
+        if node.provision_state != ACTIVE:
+            raise exceptions.CommandError(
+                "Node {0} must be in the active state".format(node.name))
+
+        baremetal_ports = ironic_client.port.list(node=node_uuid, detail=True)
+        has_free_port = False
+        for bp in baremetal_ports:
+            if 'tenant_vif_port_id' not in bp.internal_info:
+                has_free_port = True
+                continue
```

Comment:
if you have found one free port, why do you want to iterate over the rest of them? Wouldn't a `break` statement be more suitable in this case?